src/diffusers/textual_inversion_utils.py

```diff
@@ -0,0 +1,101 @@
+import torch
+
+
+class TextualInversionMixin:
+    r"""
+    Mixin class for adding textual inversion tokens and embeddings to the tokenizer and text encoder with method:
+    - [`~TextualInversionMixin.load_textual_inversion_embeddings`]
+    - [`~TextualInversionMixin.add_textual_inversion_embedding`]
+
+    Class attributes:
+    - **textual_inversion_tokens** (`List[str]`): list of tokens added to the tokenizer's vocabulary and the text
+      encoder's embedding matrix
+    """
+    textual_inversion_tokens = []
+
+    def load_textual_inversion_embeddings(self, embeddings):
```

Comment:
It would be important to also allow the possibility to directly load from the Hub. Essentially this method should allow the user to load any of the following 2000+ checkpoitns: https://huggingface.co/sd-concepts-library

E.g. the following API needs to work:

```python
pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4")
pipe.load_textual_inversion_embeddings("sd-concepts-library/midjourney-style", concept_token="<midjourney-style>")  # or just `pipe.load_textual_inversion_embeddings("sd-concepts-library/midjourney-style")` 
```

Reply:
So we would also need all the hub/folder loading functionality of: https://github.com/huggingface/diffusers/blob/3d2f24b0997ed7e5370d65e3dfbba1e27a9da9ea/src/diffusers/loaders.py#L147


Happy to go into the PR and help you! :slightly_smiling_face:

Reply:
Let's do it then. I can address this change request by myself, but your help is (always) much appreciated.